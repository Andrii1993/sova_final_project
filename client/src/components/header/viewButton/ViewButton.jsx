
import { ViewButtonWrapper } from './styled';
import PropTypes from 'prop-types';
const ViewButton = ({ children,idComponent, isActive}) => {

  return (
      <ViewButtonWrapper data-testid='test' id={idComponent} active={isActive}>
        {children}
      </ViewButtonWrapper>
  );
}

ViewButton.propTypes = {
  isActive: PropTypes.bool
};

ViewButton.defaultProps = {
  isActive: false,
  idComponent: ''
};

export default ViewButton;