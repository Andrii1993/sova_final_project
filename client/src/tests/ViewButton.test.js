import ViewButton from '../components/header/viewButton/ViewButton'
import '@testing-library/jest-dom';
import {render, cleanup} from "@testing-library/react";
import renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders with an id", () => {
  const{getByTestId}= render(<ViewButton idComponent='test-id' />);
  expect(getByTestId('test').id).toBe('test-id');
});


it("renders with active", () => {
  const {getByTestId} = render(<ViewButton isActive={true}/>);
  expect(getByTestId('test')).toHaveStyle('color: gold');
});


it('test SnapShot', () => {
  const buttonComp = renderer.create(<ViewButton idComponent='test-element' active={true}/>).toJSON()
  expect(buttonComp).toMatchSnapshot();
});

