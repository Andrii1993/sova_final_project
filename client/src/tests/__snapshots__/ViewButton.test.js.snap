// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SnapShot 1`] = `
<div
  className="sc-bczRLJ jUZZee"
  data-testid="test"
  id="test-element"
/>
`;
